go-controller/pkg/ovn/egressip.go

```diff
@@ -1147,12 +1147,15 @@ func (oc *Controller) deleteEgressIPAssignments(name string, statusesToRemove []
 			return err
 		}
 		for podKey, podStatus := range oc.eIPC.podAssignment {
-			delete(podStatus.egressStatuses, statusToRemove)
-			podNamespace, podName := getPodNamespaceAndNameFromKey(podKey)
-			if err := oc.eIPC.addPerPodGRSNAT(podNamespace, podName, podStatus.podIPs); err != nil {
-				return err
+			if _, ok := podStatus.egressStatuses[statusToRemove]; ok {
+				// this pod was managed by statusToRemove.EgressIP; we need to try and add its SNAT back towards nodeIP
+				delete(podStatus.egressStatuses, statusToRemove)
```

Comment:
right, this is confusing also because then we go delete the pod key from the map on line 1157 once the NAT is added back. I'm confused like in my other comment: https://github.com/ovn-org/ovn-kubernetes/pull/3175/files/19a577fff68a4bd90d7cacdb3caf70343f774784#r992726996

How we could have more than one status here, and how we know when to delete one status vs delete the entire key from the map.

Reply:
as discussed offline fixed this as two parts:
1) we delete the podKey from map only if there are 0 egressStatues for this pod.
2) the corner case handling for one pod multiple EIPs will be done as a new commit.